tdigest--1.0.0--1.0.1.sql

```diff
@@ -0,0 +1,29 @@
+CREATE OR REPLACE FUNCTION tdigest_add_double_count(p_pointer internal, p_element double precision, p_count bigint, p_compression int)
+    RETURNS internal
+    AS 'tdigest', 'tdigest_add_double_count'
+    LANGUAGE C IMMUTABLE;
+
+CREATE OR REPLACE FUNCTION tdigest_add_double_count(p_pointer internal, p_element double precision, p_count bigint, p_compression int, p_quantile double precision)
+    RETURNS internal
+    AS 'tdigest', 'tdigest_add_double_count'
+    LANGUAGE C IMMUTABLE;
+
+CREATE AGGREGATE tdigest(double precision, bigint, int) (
+    SFUNC = tdigest_add_double_count,
+    STYPE = internal,
+    FINALFUNC = tdigest_digest,
+    SERIALFUNC = tdigest_serial,
+    DESERIALFUNC = tdigest_deserial,
+    COMBINEFUNC = tdigest_combine,
+    PARALLEL = SAFE
+);
+
+CREATE AGGREGATE tdigest_percentile(double precision, bigint, int, double precision) (
+    SFUNC = tdigest_add_double_count,
+    STYPE = internal,
+    FINALFUNC = tdigest_percentiles,
+    SERIALFUNC = tdigest_serial,
+    DESERIALFUNC = tdigest_deserial,
+    COMBINEFUNC = tdigest_combine,
+    PARALLEL = SAFE
+);
```

Comment:
Now I "remember" why I did not add the "other APIs" besides I don't have a specific use case. This would need to add at least two more C functions, probably better to refactor the internal C APIs a bit. So last year, I thought it would be better to have a discussion first to see if we could reduce "cut-n-paste". Maybe we should address the refactoring and supporting the other APIs as a separate pull request?

Reply:
I think it'd be good to keep the "symmetry" of the API, i.e. if we have a new variant of `tdigest_percentile`, we should have a new version of `tdigest_percentile_of` I think. Assuming it makes sense in general, of course. But if I understand the reply correctly, it's "I don't a have a use case" and not "there's no use case".

More importantly, I think this should add a variant for the t-digest producing aggregate, i.e. something like

```
CREATE AGGREGATE tdigest(tdigest, value, count) { ... }
```

That would at least allow using the `tdigest_percentile_of(tdigest, ...)` variant to get almost the same thing.

FWIW I agree a refactoring might be necessary before adding those new functions - an internal function that would be called from the various functions, adding a centroid with either count=1 for points or some other value would be a big step in the right direction, probably.

Reply:
Thanks Tomas. And yes, you got my intention correctly.  Also agree on your API suggestion. So how do you/we want to proceed on this? If you have time or thoughts on the internal code refactoring, I can merge and revamp the public API code over it.

Reply:
I don't expect to have time to work on the refactoring any time soon, unfortunately. I can review a PR, but that's all.

Reply:
Understood. Shall we do the recommended API changes as a separate pull request? I have a dependency on Nils/Citus to integrate the new tdigest API for its PG13 based release.  I am "booked" for the next month or so but I can take a crack after that.